labelbox/client.py

```diff
@@ -1790,6 +1792,163 @@ def get_batch(self, project_id: str, batch_id: str) -> Entity.Batch:
+                            sendToAnnotateFromCatalog(input: $input) {
+                              taskId
+                            }
+                          }
+        """
+
+        destination_task_queue = self.build_destination_task_queue_input(
+            task_queue_id)
+        data_rows_query = self.build_catalog_query(data_rows)
+
+        source_model_run_id = params.get("source_model_run_id", None)
+        model_run_ontology_mapping = params.get("model_run_ontology_mapping",
+                                                None)
+        predictions_input = self.build_predictions_input(
+            model_run_ontology_mapping,
+            source_model_run_id) if source_model_run_id else None
+
+        source_project_id = params.get("source_project_id", None)
+        project_ontology_mapping = params.get("project_ontology_mapping", None)
+        annotations_input = self.build_annotations_input(
+            project_ontology_mapping,
+            source_project_id) if source_project_id else None
+
+        batch_priority = params.get("batch_priority", 5)
+        exclude_data_rows_in_project = params.get(
+            "exclude_data_rows_in_project", False)
+        override_existing_annotations_rule = params.get(
+            "override_existing_annotations_rule",
+            ConflictResolutionStrategy.KeepExisting)
+
+        res = self.execute(
+            mutation_str, {
+                "input": {
+                    "destinationProjectId":
+                        destination_project_id,
+                    "batchInput": {
+                        "batchName": batch_name,
+                        "batchPriority": batch_priority
+                    },
+                    "destinationTaskQueue":
+                        destination_task_queue,
+                    "excludeDataRowsInProject":
+                        exclude_data_rows_in_project,
+                    "annotationsInput":
+                        annotations_input,
+                    "predictionsInput":
+                        predictions_input,
+                    "conflictLabelsResolutionStrategy":
+                        override_existing_annotations_rule,
+                    "searchQuery": {
+                        "scope": None,
+                        "query": [data_rows_query]
+                    },
+                    "ordering": {
+                        "type": "RANDOM",
+                        "random": {
+                            "seed": random.randint(0, 10000)
+                        },
+                        "sorting": None
+                    },
+                    "sorting":
+                        None,
+                    "limit":
+                        None
+                }
+            })['sendToAnnotateFromCatalog']
+
+        return Entity.Task.get_task(self, res['taskId'])
+
+    @staticmethod
+    def build_annotations_input(project_ontology_mapping: Optional[Dict[str,
```

Comment:
nit: should we consider a `send_to_annotate_params.py` file to put all of the static methods?